modules/engageyaBidAdapter.js

```diff
@@ -0,0 +1,133 @@
+import {
+  BANNER,
+  NATIVE
+} from '../src/mediaTypes.js';
+
+const {
+  registerBidder
+} = require('../src/adapters/bidderFactory.js');
+const BIDDER_CODE = 'engageya';
+const ENDPOINT_URL = 'https://recs.engageya.com/rec-api/getrecs.json';
+const ENDPOINT_METHOD = 'GET';
+
+function getPageUrl() {
+  var pUrl = window.location.href;
+  if (isInIframe()) {
+    pUrl = document.referrer ? document.referrer : pUrl;
+  }
+  pUrl = encodeURIComponent(pUrl);
+  return pUrl;
+}
+
+function isInIframe() {
+  try {
+    var isInIframe = (window.self !== window.top);
+  } catch (e) {
+    isInIframe = true;
+  }
+  return isInIframe;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, NATIVE],
+  isBidRequestValid: function(bid) {
+    return bid && bid.params && bid.params.hasOwnProperty('widgetId') && bid.params.hasOwnProperty('websiteId') && !isNaN(bid.params.widgetId) && !isNaN(bid.params.websiteId);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var bidRequests = [];
+    if (validBidRequests && validBidRequests.length > 0) {
+      validBidRequests.forEach(function(bidRequest) {
+        if (bidRequest.params) {
+          var mediaType = bidRequest.hasOwnProperty('nativeParams') ? 1 : 2;
```

Comment:
I would advise using `mediaTypes.banner` or `mediaTypes.native` to check this, same goes for sizes on L46 and L49. Unless you have publishers that are setting nativeParams directly on adUnit, which I can't find in prebid.js docs anymore?

Reply:
When I used it the spec file failed (the tests) because it didn't have the parameters.
Do you know how to bypass this?

Reply:
I had something like this in mind (but I didn't have time to test):

```
          var mediaType = utils.deepAccess(bidRequest, 'mediaTypes.native') ? 1 : 2;
          var imageWidth = -1;
          var imageHeight = -1;
          var nativeSizes = utils.deepAccess(bidRequest, 'mediaTypes.native.image.sizes');
          var bannerSizes = utils.deepAccess(bidRequest, 'mediaTypes.banner.sizes');
          if (bannerSizes && bannerSizes.length > 0) {
            imageWidth = bidRequest.sizes[0][0];
            imageHeight = bidRequest.sizes[0][1];
          } else if (nativeSizes) {
            imageWidth = nativeSizes[0];
            imageHeight = nativeSizes[1];
          }
```

But if you say this does not work for some reason, then ok, we can go forward with what you currently have.

Reply:
lets leave it as is, I saw lot of implementations like this :)